python/semantic_kernel/functions/kernel_function.py

```diff
@@ -0,0 +1,488 @@
+    is_semantic: bool = Field(...)
+    stream_function: Optional[Callable[..., Any]] = Field(default=None)
+    parameters: List[KernelParameterMetadata] = Field(...)
+    function: Callable[..., Any] = Field(...)
+    plugins: Optional["KernelPluginCollection"] = Field(default=None)
+    ai_service: Optional[Union[TextCompletionClientBase, ChatCompletionClientBase]] = Field(default=None)
+    prompt_execution_settings: PromptExecutionSettings = Field(default_factory=PromptExecutionSettings)
+    chat_prompt_template: Optional[ChatPromptTemplate] = Field(default=None)
+
+    def __init__(
+        self,
+        function: Callable[..., Any],
+        parameters: List[KernelParameterMetadata],
+        description: str,
+        plugin_name: str,
+        function_name: str,
+        is_semantic: bool,
+        delegate_stream_function: Optional[Callable[..., Any]] = None,
+        **kwargs: Dict[str, Any],
+    ) -> None:
+        """
+        Initializes a new instance of the KernelFunction class
+
+        Args:
+            delegate_type (DelegateTypes): The delegate type for the function
+            delegate_function (Callable[..., Any]): The delegate function for the function
+            parameters (List[ParameterView]): The parameters for the function
+            description (str): The description for the function
+            plugin_name (str): The name of the plugin
+            name (str): The name of the function
+            is_semantic (bool): Whether the function is semantic
+            delegate_stream_function (Optional[Callable[..., Any]]): The delegate stream function for the function
+            kwargs (Dict[str, Any]): Additional keyword arguments
+        """
+        chat_prompt_template = kwargs.pop("chat_prompt_template", None)
+
+        super().__init__(
+            function=function,
+            parameters=parameters,
+            description=description,
+            plugin_name=plugin_name,
+            name=function_name,
+            is_semantic=is_semantic,
+            stream_function=delegate_stream_function,
+            chat_prompt_template=chat_prompt_template,
+        )
+
+    @staticmethod
+    def from_native_method(method: Callable[..., Any], plugin_name: str) -> "KernelFunction":
+        """
+        Create a KernelFunction from a native method.
+
+        Args:
+            method (Callable[..., Any]): The method to create the function from
+            plugin_name (str): The name of the plugin
+
+        Returns:
+            KernelFunction: The kernel function
+        """
+        if method is None:
+            raise ValueError("Method cannot be `None`")
+
+        if not hasattr(method, "__kernel_function__") or method.__kernel_function__ is None:
+            raise ValueError("Method is not a Kernel function")
+
+        parameters = []
+        # kernel_function_context_parameters are optionals
+        if hasattr(method, "__kernel_function_context_parameters__"):
```

Comment:
This will never be true now, right? Since we don't have kernel_function_context_parameters.